### PR TITLE
Use ES6 modules in rally.js

### PR DESCRIPTION
**Pull request by Dexterp37**
This additionally performs some logistical tasks:

- Add a separate changelog for rally.js, to make it easier to understand what's going on for rally.js consumers.
- Add documentation for doing manual testing of rally.js.

Checklist for reviewer:

- [ ] The description should reference a bug or github issue, if relevant.
- [ ] There must be a [`CHANGELOG.md`](./CHANGELOG.md) entry for any non-test change.
- [ ] Any change to the NPM commands must be carefully reviewed to make sure it won't break the Add-ons pipeline.
- [ ] Any version increase must follow the [release process](./docs/RELEASE_PROCESS.md).

### CODE REVIEW

**Comment by Dexterp37** [comment]
@brizental would you kindly review this?

File: package.json [hunk 1]
```diff
@@ -16,7 +16,7 @@
     "test": "npm run --silent skip-taskcluster || (npm run test-addon && npm run test-support-library)",
     "test-addon": "npm run --silent skip-taskcluster || (mocha --require  \"./tests/hooks.js\" \"./tests/core-addon/unit/*.js\")",
     "test-integration": "npm run --silent skip-taskcluster || (npm run build && mocha --timeout 30000 \"./tests/core-addon/integration/*.js\")",
-    "test-support-library": "npm run --silent skip-taskcluster || (mocha --require  \"./tests/hooks.js\" \"./tests/support/*.js\")"
+    "test-support-library": "npm run --silent skip-taskcluster || (mocha --require  \"./tests/hooks.js\" \"./tests/support/*.mjs\")"
```
Comment (brizental) [inline package.json:19]: Is it really necessary to change the extension of the test files?

Comment (brizental) [inline package.json:19]: Is it really necessary to change the extension of the test files?

**Review by brizental — COMMENTED** [review]
r+wc

File: package.json (hunk 1 above)
Comment (Dexterp37) [reply package.json:19]: Yes, if we don't, then `SyntaxError: Cannot use import statement outside a module` :-D

**Pull request merged**